dcos/package.py

```diff
@@ -845,7 +846,15 @@ def copy_to_cache(self, target_dir):
                 tmp_file = os.path.join(tmp_dir, 'packages.zip')
 
                 # Download the zip file.
-                urllib.request.urlretrieve(self.url, tmp_file)
+
+                req = requests.get(self.url, stream=True,
+                                   proxies=util.get_proxy_dict())
+                if req.status_code == 200:
+                    with open(tmp_file, 'wb') as f:
+                        for chunk in req.iter_content(1024):
+                            f.write(chunk)
+                else:
+                    raise Exception
```

Comment:
DCOSException is re-raised below 
        except Exception:
            raise DCOSException(
                'Unable to fetch packages from [{}]'.format(self.url))